src/ml/trainer/JasmineGraphTrainingSchedular.cpp

```diff
@@ -177,3 +178,215 @@ JasmineGraphTrainingSchedular::packPartitionsToMemory(vector<pair<int, int>> par
+
+        vector<vector<int>> partitionDetails; //Vector for node count, edge count, feature count for memory estimation of each partition
+        map<int, int> partitionWorkerMap;
+        //Iterate through partition list and get partition details
+        for (std::vector<vector<pair<string, string>>>::iterator i = results.begin(); i != results.end(); ++i) {
+            std::vector<pair<string, string>> rowData = *i;
+            id_partition = rowData.at(0).second;
+            vertexCount = rowData.at(1).second;
+            centralvertexCount = rowData.at(2).second;
+            edgeCount = rowData.at(3).second;
+            centralEdgeCount = rowData.at(4).second;
+
+            //Compute partition metrics
+            partitionID = stoi(id_partition);
+            vertexcount = stoi(vertexCount) + stoi(centralvertexCount);
+            edgecount = stoi(edgeCount) + stoi(centralEdgeCount);
+
+            vector<int> partitionCounts; //Vector for partition node count, edge count and feature count
+            partitionCounts.push_back(partitionID);
+            partitionCounts.push_back(vertexcount);
+            partitionCounts.push_back(edgecount);
+            partitionCounts.push_back(featurecount);
+
+            partitionDetails.push_back(partitionCounts);
+
+            //Store assigned worker per partition
+            workerid = stoi(rowData.at(5).second);
+            partitionWorkerMap[partitionID] = workerid;
+
+        }
+
+        long availableMemory = getAvailableMemory(j->second); //Host memory
+
+        //Get memory estimation list for each partition of host
+        vector<pair<int, int>> partitionMemoryList = estimateMemoryDistOpt(partitionDetails, availableMemory);
+
+        //Get schedule for host
+        map<int, map<int, int>> scheduledPartitionSets = bestFitPartition(partitionMemoryList, partitionWorkerMap, availableMemory);
+        scheduleForEachHost.insert(make_pair(j->second, scheduledPartitionSets));
+    }
+    return scheduleForEachHost;
+}
+
+/** Estimate memory of graph partitions using the partition details
+ *
+ * @param partitionDetails a vector of vectors containing partition ID, node count, edge count, feature count per partition
+ * @param availableMemory total memory of host
+ */
+vector<pair<int, int>>
+JasmineGraphTrainingSchedular::estimateMemoryDistOpt(vector<vector<int>> partitionDetails, long availableMemory) {
+
+
+    vector<pair<int, int>> partitionMemoryList; //Vector of estimated sizes of partitions
+
+    //Iterate through partitions and estimate sizes
+    for (vector<vector<int>>::iterator i = partitionDetails.begin(); i != partitionDetails.end(); i++) {
+        int partitionID = (*i)[0];
+        int nodeCount = (*i)[1];
+        int edgeCount = (*i)[2];
+        int featureCount = (*i)[3];
+
+        int graphSize = 8*nodeCount*edgeCount*featureCount / (1024 * 1024); //Size in hard disk
+
+        int sizeInMem = 36 * graphSize + 2; //Size when loaded into memory
```

Comment:
Why we have `36` here? How did we choose that value?

Reply:
Results from experiments to map the relationship between HDD size and size in memory. However, there is a typo here and the value was changed to 3.6

Reply:
Can you add one comment line describing how 3.6 was derived?

Reply:
Comment added

Reply:
Ok